### DISCUSSION TITLE
Not able to setup AWS S3 For Storage

### DISCUSSION THREAD

**Discussion by manishonc** [pocketbase/pocketbase#2975] [Q&A] [answered]
I am trying to setup AWS S3 in pocketbase
But it's showing  me my access key ! No Such Key I have included the screenshot

![IMG_20230723_002757.jpg](https://github.com/pocketbase/pocketbase/assets/84983048/a4259122-8388-475f-9329-9db2652c6308)

I even Created new Key Access key and Secret Still facing the Same issue.

**Comment by ganigeorgiev** [comment, marked as answer]
You can check https://github.com/pocketbase/pocketbase/discussions/679.

Your "Endpoint" address doesn't look correct. In your case  I think it should be `s3.us-east-2.amazonaws.com`.

Also you may not need the `Force path-style addressing` check.

**Reply by manishonc** [reply]
Ok let me try thanks for quick reply

**Reply by robitar** [reply]
@ganigeorgiev just to note that it might be worth putting this somewhere near that field, say as a tooltip (the expected format for endpoint).

If you are not familiar with S3 it might not be obvious, and that you should omit the https etc. I ended up here after trying a few things and scanning the docs.

**Reply by ganigeorgiev** [reply]
@robitar The endpoint address depends on the S3 vendor so I'm not sure what exactly to clarify. Regarding the https part - it should be fine to start with https. This is an old discussion and I kindof already forgot the details but I think the the original issue is that the endpoint contains the bucket name.

**Reply by robitar** [reply]
Yea thats what I figured when trying to work it out, that it could be any vendor. I guess my only suggestion would be to just add a tooltip note on the assumption that you are using AWS, along the lines of:

> If using AWS, this will typically be `s3.[REGION].amazonaws.com`, for other providers check their documentation

There's already an assumption of AWS on the tooltip for 'Force path-style addressing', so there's some precedent there.

As for excluding bucket name, that perhaps adds to the confusion, as most examples you will see do in fact include the bucket name in the URI. I guess it should be obvious that you leave it out because then you wouldn't have another field requesting it, but then again, you need to put in the region in both places!

**Reply by ganigeorgiev** [reply]
No sorry, I don't think it is necessary and it could actually cause more confusion than be helpful. Again, the endpoint address is entirely vendor dependent and whether it will contain the region or anything else would vary from one service to another. Users are expected to always refer to the vendors docs. Specifically for the AWS see https://docs.aws.amazon.com/general/latest/gr/s3.html.

**Reply by robitar** [reply]
Yea that's a fair point, even with AWS it's hardly straightforward, and I suppose you should be checking up on the docs as you go anyway.

**Comment by manishonc** [comment]
> I am trying to setup AWS S3 in pocketbase But it's showing me my access key ! No Such Key I have included the screenshot
> 
> ![IMG_20230723_002757.jpg](https://user-images.githubusercontent.com/84983048/255346935-a4259122-8388-475f-9329-9db2652c6308.jpg)
> 
> I even Created new Key Access key and Secret Still facing the Same issue.

I need to solve this urgently. Please help me

**Reply by ganigeorgiev** [reply]
Please check the above reply.

**Comment by manishonc** [comment]
> You can check https://github.com/pocketbase/pocketbase/discussions/679.
> 
> Your "Endpoint" address doesn't look correct. In your case  I think it should be `s3.us-east-2.amazonaws.com`.
> 
> Also you may not need the `Force path-style addressing` check.

It's Connected Now Thanks .
 
Thanks Love you man 💗
